lib/collection.js

```diff
@@ -395,56 +402,37 @@ Collection.prototype.find = deprecateOptions(
     // Build the find command
     const findCommand = {
       find: this.s.namespace,
-      limit: newOptions.limit,
-      skip: newOptions.skip,
+      limit: finalOptions.limit,
+      skip: finalOptions.skip,
       query: selector
     };
 
-    // Ensure we use the right await data option
-    if (typeof newOptions.awaitdata === 'boolean') {
-      newOptions.awaitData = newOptions.awaitdata;
+    // Merge in options to command
+    for (let name in finalOptions) {
+      if (finalOptions[name] != null && name !== 'session') {
+        findCommand[name] = finalOptions[name];
+      }
     }
 
-    // Translate to new command option noCursorTimeout
-    if (typeof newOptions.timeout === 'boolean') newOptions.noCursorTimeout = newOptions.timeout;
-
-    decorateCommand(findCommand, newOptions, ['session', 'collation']);
```

Comment:
are we fully accounting for command decoration in the code? I see you have code over there to merge options from `finalOptions` into the command, but only if the key isn't `session`. Is that parity with this?

Reply:
Good catch